scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import division, print_function, absolute_import
+
+__all__ = ['geometric_slerp']
+
+import numpy as np
+from scipy.spatial.distance import euclidean
+
+def _geometric_slerp(start, end, t):
+    # create an orthogonal basis using QR decomposition
+    basis = np.vstack([start, end])
+    Q, R = np.linalg.qr(basis.T)
+    signs = np.sign(np.diag(R))
```

Comment:
In the case of degenerate input, this sign computation destroys the relationship between the `basis` variable and its QR decomposition. Sorry that I did not think of this.

If we replace it by:
`    signs = 2 * (np.diag(R) >= 0) - 1`
(which effectively disallows zero signs) the QR relationship is preserved. This also means that degenerate input no longer goes through the origin and looks like a normal slerp.

Reply:
I often have a need to correct the signs of a QR decomposition. From wikipedia:
>...the factorization is unique if we require the diagonal elements of R to be positive.

@tylerjereddy I have often thought that this would be worth fixing upstream, i.e. in numpy. Do you think they would be receptive to that?

Reply:
Probably best to bring that up with the NumPy team, I've applied your patch here & adjusted tests accordingly.

Reply:
I made an issue: https://github.com/numpy/numpy/issues/15628